### ISSUE TITLE
vaadin-combo-box: drop-down content unwrap

### ISSUE THREAD

**Issue by prashant7526** [vaadin/vaadin-combo-box#892]
Description
Vaadin-combo-box by default wrap the drop-down-list. while It would be better to have Combobox like vaadin 8. This is irritating having a wrapped drop-down list.

Expected outcome
Unwrap and automatically adjust, until not specifically instruct to wrap (Should work like vaadin 8 combo box).

### Browsers Affected

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE 11
- [ ] iOS Safari
- [ ] Android Chrome

**Comment by rolfsmeds** [comment]
Just to check if we understood this correctly:
You want the _labels_ of the items in the combo box dropdown to not wrap, and for the dropdown's width to be automatically set according to the width of the widest item label, correct?

As a workaround, from V14.3 the dropdown's width can be set to a custom width: https://github.com/vaadin/vaadin-combo-box/issues/529

**Comment by prashant7526** [comment]
Yes, but what I was saying if I have a longer string (which we have like [company name, City, Postal Code] now how wide I have to create the Combobox, while in Vaadin 8 the drop-down was working great. I don't know what changed. Or just me feeling that way migrating from GWT component from vaadin 8 to web comp. of vaadin 14?

**Comment by rolfsmeds** [comment]
The Vaadin 8 combobox dropdown was split into discrete pages, which made auto-adjusting its width based on the longest item on the page easy. The V10+ combobox dropdown is a single scrolling page, however, which, combined with lazy loading items, makes it difficult if not impossible to automatically adjust the width based on the content. Would it be based on the first "page" of items, and maybe then still be too narrow when more items are loaded? Or would the dropdown width fluctuate as you scroll up/down?

But it's clear what you're asking for, so I suggest changing the issue title to be a bit more descriptive, e.g. "Dropdown width automatically based on item labels" or something like that.

**Comment by prashant7526** [comment]
So, will this combo-box will remain like this forever? I mean, it is just wired (not for me) but for my clients, they need more info to identify or pick a specific person/or his/her detail. For example in laymen language: If I am working with three different garages with same name but in a different location, in that case, I am loading my full name, garage name, city, postal code. And in no way, it is going to be in one line (in V10+ Conbobox). I don't know, I just gave you the real scenario where I am stuck.

**Comment by web-padawan** [comment]
There are technical reasons why the items behave like that mentioned at https://github.com/vaadin/vaadin-combo-box/issues/529#issuecomment-330756964

> we are using iron-list, and that positions the items absolutely, so the width of the items do not affect the width of the overlay. So we would still have to measure the width of the items to be able to make the overlay adjust to it.

We might fix that in a similar way as we did for grid, by adding `auto-width` property: https://github.com/vaadin/vaadin-grid/pull/1595

**Comment by prashant7526** [comment]
Thank you guys, hoping the new one will come out soon.

**Comment by web-padawan** [comment]
I need to clarify that this is currently not in our short-term roadmap. We would need to do some prototyping first.

Compared to grid, there is an additional challenge related to the fact that items are only rendered after combo-box is opened.

**Comment by rolfsmeds** [comment]
>For example in laymen language: If I am working with three different garages with the same name but in a different location, in that case, I am loading my full name, garage name, city, postal code. And in no way, it is going to be in one line...

In these types of cases I think it would make a lot of sense to use a custom renderer where the location is in a separate string below the garage name, see e.g. https://vaadin.com/components/vaadin-combo-box/java-examples/presentation. Also from 14.3 onward you can set a custom width for the dropdown using the `--vaadin-combo-box-overlay-width` css variable.

**Comment by prashant7526** [comment]
Oh cool, will try to implement this one. Thanks for the suggestion.
